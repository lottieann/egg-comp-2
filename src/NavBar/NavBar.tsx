import * as React from 'react'

import Link from 'next/link'

import styled from 'styled-components'

export interface NavBarProps {
  currentPage: string
}

const StyledContainer = styled.nav`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(205, 202, 202, 0.3);
  width: 100vw;
  height: 15vh;
  position: absolute;
  z-index: 20;
  top: 0;
  left: 0;
`

const StyledList = styled.ul`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  list-style-type: none;
  margin: 10px;
  padding: 0;
  width: 45vw;

  @media screen and (max-width: 600px) {
    width: 100vw;
  }
`
const StyledListItem = styled.li`
  padding: 15px;
`
const StyledAnchor = styled.a`
  text-decoration: none;
  color: black;
  cursor: pointer;
  font-weight: 700;

  &:hover {
    font-weight: 900;
  }

  ${(props: { currentPage: boolean }) =>
    props.currentPage ? 'text-decoration: underline;' : null}
`

export const NavBar = (props: NavBarProps) => {
  return (
    <StyledContainer>
      <StyledList>
        <StyledListItem>
          <Link href="/">
            <StyledAnchor
              data-testid="homepage-link"
              currentPage={props.currentPage === 'HomePage'}
            >
              Home
            </StyledAnchor>
          </Link>
        </StyledListItem>
        <StyledListItem>
          <Link href="/schedule">
            <StyledAnchor currentPage={props.currentPage === 'Schedule'}>
              Schedule
            </StyledAnchor>
          </Link>
        </StyledListItem>
        <StyledListItem>
          <Link href="/gallery">
            <StyledAnchor currentPage={props.currentPage === 'Gallery'}>
              Gallery
            </StyledAnchor>
          </Link>
        </StyledListItem>
      </StyledList>
    </StyledContainer>
  )
}
