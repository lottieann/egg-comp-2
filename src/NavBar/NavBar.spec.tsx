import * as React from 'react'

import 'jest-styled-components'

import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { NavBar, NavBarProps } from './NavBar'

const setup = (props: NavBarProps) => {
  return <NavBar {...props} />
}

describe('NavBar', () => {
  it.skip('should redirect to correct link', () => {
    const { getByTestId } = render(
      setup({
        currentPage: 'HomePage',
      }),
    )
    fireEvent.click(getByTestId('homepage-link'))
    const target = 'http://localhost/'
    expect(window.location.href).toBe(target)

  })
  it('should render NavBar correctly', () => {
    const { container } = render(
      setup({
        currentPage: 'HomePage',
      }),
    )
    expect(container).toMatchSnapshot()
  })
})
