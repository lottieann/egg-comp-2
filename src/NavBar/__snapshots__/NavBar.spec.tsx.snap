// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render NavBar correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(205,202,202,0.3);
  width: 100%;
  height: 15vh;
  position: absolute;
  z-index: 20;
  top: 0;
  left: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  list-style-type: none;
  margin: 10px;
  padding: 0;
  width: 35vw;
}

.c2 {
  padding: 15px;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  cursor: pointer;
  font-weight: 700;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:hover {
  font-weight: 900;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  cursor: pointer;
  font-weight: 700;
}

.c4:hover {
  font-weight: 900;
}

<div>
  <nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          class="c3"
          data-testid="homepage-link"
        >
          Home
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          class="c4"
        >
          Submit
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          class="c4"
        >
          Gallery
        </a>
      </li>
      <li
        class="c2"
      >
        <a
          class="c4"
        >
          Vote!
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
